rasterio/_crs.pyx

```diff
@@ -170,24 +170,43 @@ cdef class _CRS(object):
         int
 
         """
-        cdef OGRSpatialReferenceH osr = NULL
+        if self._epsg is not None:
+            return self._epsg
+
+        auth = self.to_authority()
+        if auth is None:
+            return None
+        name, code = auth
+        if name.upper() == "EPSG":
+            self._epsg = int(code)
+        return self._epsg
 
-        if self._epsg is None:
-            try:
-                osr = exc_wrap_pointer(OSRClone(self._osr))
-                exc_wrap_ogrerr(OSRMorphFromESRI(osr))
-                if OSRAutoIdentifyEPSG(osr) == 0:
-                    epsg_code = OSRGetAuthorityCode(osr, NULL)
-                    if epsg_code != NULL:
-                        self._epsg = int(epsg_code.decode('utf-8'))
-                    else:
-                        self._epsg = None
-                else:
-                    self._epsg = None
-            finally:
-                _safe_osr_release(osr)
+    def to_authority(self):
+        """The authority name and code of the CRS
 
-        return self._epsg
+        Returns
+        -------
+        (str, str) or None
```

Comment:
@snowman2 yes! I was going to ask about str vs int codes and realized that str follows the WKT example, where codes are strings.

Reply:
Yep, one example of where a string code is used: `OGC:CRS84`